Core/include/Acts/EventData/MultiTrajectory.hpp

```diff
@@ -55,6 +55,34 @@ namespace detail_lt {
 template <typename T, bool select>
 using ConstIf = std::conditional_t<select, const T, T>;
 
+/// Helper type to make a member pointers constness transitive.
+template <typename T>
+struct TransitiveConstPointer {
```

Comment:
I would actually prefer to put this into a `class` with private member data, since the possibility do directly access the `ptr` kind of defeats its purpose...

Would maybe also be good to put this into `Utilities/Core`

Reply:
I'll make it a class and make the member private, but I think I'd keep this in the detail namespace in this file for now.